WordPress/gutenberg
# Try: bundle WP_REST_Global_Styles_Controller class instead of inheriting per WordPress version

## What?
Similar to:

- https://github.com/WordPress/gutenberg/pull/46579

This PR bundles rest global styles controller code into one class `WP_REST_Global_Styles_Controller` to avoid conflicts, and to make sure the response always reflects changes from `WP_Theme_JSON_Gutenberg`.

The `WP_REST_Global_Styles_Controller` is the endpoint that fetches the contents of the theme global styles custom post from the database.

## Why?
Context for change: https://github.com/WordPress/gutenberg/pull/53451/files#r1289969201

For many of the reasons cited in https://github.com/WordPress/gutenberg/pull/46579, but also because the underlying response is directly affected by changes to `WP_Theme_JSON_Gutenberg`.

## How?
Removing all extensions in `lib/compat` and creating an "evergreen" file under `/lib`.

## Testing Instructions
Check that all tests pass on CI.

Ensure both classic (TwentyTwenty) and block themes (TwentyTwentyThree) load properly and that any presets are loaded and work correctly. 

For block themes, perform some common global styles-related functions, e.g., 

In the site editor, check that the endpoint returns the current merged theme.json. Run this command in the browser console: 

```js
await wp.apiFetch( { url: `/index.php?rest_route=%2Fwp%2Fv2%2Fglobal-styles%2F${ wp.data.select('core').__experimentalGetCurrentGlobalStylesId() }&_locale=user` } );
```

In the site editor, make some changes to the global styles (e.g.,: select a different style variation, add a custom color palette). Refresh the page and verify they are applied everywhere (editor, front-end).

Check that the revisions count in the global styles revisions drop down is accurate
